### PR TITLE
Ensure that calls to superset and openSRP API endpoints are done in order.

### LINKED ISSUE #157: TypeError: can not read undefined property
Screenshot from 2020-01-13 11-29-53

### PR DESCRIPTION
**Pull request by FrankApiyo**
Make `fetchData`,  the function responsible from getting data from superset to the browser an **async** function and prepend all calls to `supersetFetch` with `await`.

Fixes #157

### CODE REVIEW

File: src/helpers/utils.tsx [hunk 1]
```diff
@@ -509,7 +510,8 @@ export function getLinkToPatientDetail(smsData: SmsData, prependWith: string) {
  * c. LOCATION_SLICES
  * d. SUPERSET_SMS_DATA_SLICE
  */
-export async function fetchData() {
+// tslint:disable-next-line: no-shadowed-variable
+export async function fetchData(supersetFetchMethod: any = supersetFetch) {
```
Comment (moshthepitt) [inline src/helpers/utils.tsx]: This seems like a rule that should not be disabled.  Why do it?

File: src/helpers/utils.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { getOnadataUserInfo, getOpenSRPUserInfo } from '@onaio/gatekeeper';
 import { SessionState } from '@onaio/session-reducer';
+import superset from '@onaio/superset-connector/dist/types';
```
[the remark this answers is not in the document]
Comment (FrankApiyo) [reply src/helpers/utils.tsx]: removed

File: src/helpers/utils.tsx (hunk 1 above)
Comment (moshthepitt) [reply src/helpers/utils.tsx]: Please dont use `any` here.  You clearly already know what you are expecting, no?

Comment (moshthepitt) [reply src/helpers/utils.tsx]: > This seems like a rule that should not be disabled. Why do it?

@FrankApiyo this has not been addressed

Comment (FrankApiyo) [reply src/helpers/utils.tsx]: @moshthepitt this was addressed here :point_right: https://github.com/OpenSRP/opensrp-web/pull/158/commits/1ef05b9c2fa22fcbc694b2dc558ceba54fe9042e

File: src/helpers/utils.tsx [hunk 3]
```diff
@@ -508,35 +508,36 @@ export function getLinkToPatientDetail(smsData: SmsData, prependWith: string) {
  * b. USER_LOCATION_DATA_SLICE
```
Comment (moshthepitt) [inline src/helpers/utils.tsx]: Please fix typos in this file, including:

- hierarchichal
- supserset
- commnue
- distticts
- provices
- Miliseconds

Comment (FrankApiyo) [reply src/helpers/utils.tsx]: thanks for poiting this out for me; I just installed (code-spell-checker)[https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker] to help me with this.

**Pull request merged**